app/main.py

```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+def create_person_list(people):
+    for person in people:
+        Person(person["name"], person["age"])
+
+    for person in people:
+        if "wife" in person and person["wife"]:
+            Person.people[person["name"]].wife = Person.people[person["wife"]]
+        elif "husband" in person and person["husband"]:
+            Person.people[person["name"]].husband = Person.people[person["husband"]]
```

Comment:
The code assumes that the 'wife' or 'husband' referenced in the `people` dictionary already exists. If a person is referenced as a spouse but has not been created yet, this will raise a `KeyError`. Ensure that all referenced people are created before establishing relationships, or add error handling to manage such cases.